### PR TITLE
NAS-104333 / 12.0 / Cross-platform services

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

**Comment by themylogin** [comment]
@william-gr @sonicaj it's only half of the job (the easiest half) - refactoring and changing interfaces to be ready to be cross-platform. I think it will be convenient to review and merge this part first, and then I'll work on Linux implementations (which will be another giant PR), are we good with that?

**Comment by william-gr** [comment]
Sounds good to me! I will probably end up reviewing this early next week though, its a big one and i dont want to rush it.

**Comment by sonicaj** [comment]
It sounds good to me as well

File: src/middlewared/middlewared/plugins/service_/services/iscsitarget.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    name = "iscsitarget"
+    reloadable = True
+
+    etc = ["ctld"]
+
+    freebsd_rc = "ctld"
+    freebsd_pidfile = "/var/run/ctld.pid"
+
+    async def before_stop(self):
+        if platform.system() == "FreeBSD":
```
[the remark this answers is not in the document]
Comment (themylogin) [reply src/middlewared/middlewared/plugins/service_/services/iscsitarget.py]: Will fix in another PR

**Pull request merged**